presto-main/src/main/java/com/facebook/presto/operator/PipelineContext.java

```diff
@@ -492,6 +507,18 @@ else if (driverContext.isFullyBlocked()) {
             }
         }
 
+        int queuedDrivers;
+        if (partitioned) {
+            queuedDrivers = totalSplits - runningDrivers - blockedDrivers - completedDrivers;
+            if (queuedDrivers < 0) {
```

Comment:
how about `if (queuedDrivers < physicallyQueuedDrivers) queuedDrivers = physicallyQueuedDrivers`?

Reply:
I think that's a good idea.

I do have a concern about hiding bugs. But I think I have this problem here regardless of this change anyways.

Reply:
I'm more concerned about observation skew due to the lack of a snapshot.  I think that this provides a better answer than 0.